# Advent of Code Day 1

In [1]:
import pandas as pd

### setting up input.txt into df

In [2]:
df = pd.read_csv('input.txt')

In [3]:
df = df.map.str.split('', expand=True)

In [4]:
df = df.drop(columns=[0,32], axis=1)

In [5]:
df = df.replace('.',0)
df = df.replace('#',1)

In [6]:
df

1   2   3   4   5   6   7   8   9   10  ...  22  23  24  25  26  27  28  \
0     0   0   0   0   0   0   1   0   0   0  ...   0   1   0   0   0   0   0   
1     0   1   0   0   0   0   0   1   0   0  ...   0   0   0   0   0   1   1   
2     0   0   0   0   0   0   1   0   1   0  ...   0   0   0   0   0   0   0   
3     0   0   0   0   0   0   0   0   0   0  ...   0   0   0   1   0   0   0   
4     0   0   0   0   0   1   0   1   0   0  ...   1   0   0   1   0   0   1   
..   ..  ..  ..  ..  ..  ..  ..  ..  ..  ..  ...  ..  ..  ..  ..  ..  ..  ..   
318   1   1   0   1   0   1   0   0   0   0  ...   1   0   1   1   1   1   0   
319   0   0   0   0   0   0   0   0   0   0  ...   0   0   0   1   0   0   0   
320   0   0   0   1   1   0   0   0   0   0  ...   0   0   1   0   1   0   0   
321   0   0   1   0   0   0   0   0   0   0  ...   0   0   0   1   0   0   0   
322   0   0   1   0   0   1   0   0   0   1  ...   0   0   0   0   0   1   0   

     29  30  31  
0     0   0   0  
1     0   0   0  
2     0   0   0  
3     0   0   0  
4     0   0   1  
..   ..  ..  ..  
318   0   0   0  
319   0   0   0  
320   0   0   1  
321   0   0   0  
322   0   0   0  

[323 rows x 31 columns]

### adding the same df together to expand the 'map'

In [13]:
df = pd.concat([df,df], axis=1)

In [14]:
df = pd.concat([df,df], axis=1)

In [15]:
df = pd.concat([df,df], axis=1)

In [17]:
df = pd.concat([df,df], axis=1)

In [19]:
df = pd.concat([df,df], axis=1)

In [80]:
df = pd.concat([df,df], axis=1)

In [82]:
df.columns = range(0,3968)

In [83]:
df

0     1     2     3     4     5     6     7     8     9     ...  3958  \
0       0     0     0     0     0     0     1     0     0     0  ...     0   
1       0     1     0     0     0     0     0     1     0     0  ...     0   
2       0     0     0     0     0     0     1     0     1     0  ...     0   
3       0     0     0     0     0     0     0     0     0     0  ...     0   
4       0     0     0     0     0     1     0     1     0     0  ...     1   
..    ...   ...   ...   ...   ...   ...   ...   ...   ...   ...  ...   ...   
318     1     1     0     1     0     1     0     0     0     0  ...     1   
319     0     0     0     0     0     0     0     0     0     0  ...     0   
320     0     0     0     1     1     0     0     0     0     0  ...     0   
321     0     0     1     0     0     0     0     0     0     0  ...     0   
322     0     0     1     0     0     1     0     0     0     1  ...     0   

     3959  3960  3961  3962  3963  3964  3965  3966  3967  
0       1     0     0     0     0     0     0     0     0  
1       0     0     0     0     1     1     0     0     0  
2       0     0     0     0     0     0     0     0     0  
3       0     0     1     0     0     0     0     0     0  
4       0     0     1     0     0     1     0     0     1  
..    ...   ...   ...   ...   ...   ...   ...   ...   ...  
318     0     1     1     1     1     0     0     0     0  
319     0     0     1     0     0     0     0     0     0  
320     0     1     0     1     0     0     0     0     1  
321     0     0     1     0     0     0     0     0     0  
322     0     0     0     0     1     0     0     0     0  

[323 rows x 3968 columns]

# Part 1

In [46]:
# counts trees with right 3, down 1
count = 0
tree = 0

for y in df.index:
    
    if count % 3 == 0:
        
        if df[count][y] == 1:
            
            tree +=1
            
    count+=3
    
tree

259

# Part 2
- Right 1, down 1. 64
- Right 3, down 1. 259
- Right 5, down 1. 65
- Right 7, down 1. 59
- Right 1, down 2. 35

In [67]:
# right 1, down one
count = 0
tree = 0

for y in df.index:
       
    if df[count][y] == 1:
            
            tree +=1
            
    count+=1
    
tree, count

(64, 323)

In [69]:
# right 5, down 1
count = 0
verify = 0
tree = 0

for y in df.index:
    
    if count % 5 == 0:
        
        if df[count][y] == 1:
            
            tree +=1
            
    count+=5
    verify +=1
    
tree, verify

(65, 323)

In [84]:
# right 7, down 1
count = 0
tree = 0
verify = 0

for y in df.index:
        
    if count % 7 == 0:
        
        if df[count][y] == 1:
            
            tree +=1
        
    count+=7
    
tree, count

(59, 2261)

In [85]:
# right 1, down 2
count = 0
tree = 0

for y in df.index:
        
    if y % 2 != 0:
        tree+=0
        
    elif y % 2 == 0:
        if df[count][y] == 1:
            tree +=1
        
        count+=1
    
tree, count

(35, 162)